modules/mapping-utils/src/createConnectionResolvers.js

```diff
@@ -3,28 +3,36 @@ import { get } from 'lodash';
 
 import resolveAggregations from './resolveAggregations';
 import resolveHits from './resolveHits';
+import { fetchMapping } from './utils/fetchMapping';
+import loadExtendedFields from './utils/loadExtendedFields';
 
 type TcreateConnectionResolversArgs = {
   type: Object,
 };
 type TcreateConnectionResolvers = (
   args: TcreateConnectionResolversArgs,
 ) => Object;
+
 let createConnectionResolvers: TcreateConnectionResolvers = ({
   type,
   indexPrefix,
   createStateResolvers = true,
   Parallel,
 }) => ({
   [type.name]: {
-    mapping: () => {
-      // TODO: stitch extended mapping
-      return type.mapping;
+    mapping: async (obj, { indices }, { es, projectId }) => {
+      const { index, es_type: esType } = type;
+      const { mapping: esMapping } = await fetchMapping({ index, esType, es });
+      const mappings =
+        esMapping[Object.keys(esMapping)[0]].mappings[esType].properties;
+      return mappings;
     },
-    extended: (obj, { fields }) => {
+    extended: async (obj, { fields }, { es, projectId }) => {
+      const { index } = type;
+      const extendedFields = await loadExtendedFields({ es, projectId, index });
       return fields
-        ? type.extendedFields.filter(x => fields.includes(x.field))
-        : type.extendedFields;
+        ? extendedFields.filter(x => fields.includes(x.field))
+        : extendedFields;
```

Comment:
this removes the in memory caching of mapping and extended mapping of the search api